filters/filter.go

```diff
@@ -22,103 +22,171 @@ package filters
 
 import (
 	"bytes"
+	"errors"
 	"fmt"
-	"sort"
+	"strings"
 )
 
-// logicalOp is a logical operator
-type logicalOp string
+var (
+	errInvalidFilterPattern = errors.New("invalid filter pattern defined")
+)
+
+// LogicalOp is a logical operator
+type LogicalOp string
 
 // A list of supported logical operators
 const (
-	conjunction logicalOp = "&&"
-	disjunction logicalOp = "||"
+	// Conjunction is logical AND
+	Conjunction LogicalOp = "&&"
+	// Disjunction is logical OR
+	Disjunction LogicalOp = "||"
+	// WildcardChar is the wildcard character
+	WildcardChar = "*"
 )
 
-// IDFilter matches an id against certain conditions
-type IDFilter interface {
+// Filter matches a string against certain conditions
+type Filter interface {
 	fmt.Stringer
 
 	// Matches returns true if the conditions are met
-	Matches(id string) bool
+	Matches(val string) bool
+}
+
+// NewFilter supports startsWith, endsWith, contains and a single wildcard
+// TODO(martinm): add rest of glob matching support and negation
+func NewFilter(pattern string) (Filter, error) {
```

Comment:
Yes, we use this inside rules right now and will do in other places in the stack moving forward. There are lots of use cases outside of rules, like blacklisting/whitelisting etc.